octopoes/bits/missing_dkim/missing_dkim.py

```diff
@@ -1,19 +1,28 @@
-from typing import Dict, Iterator, List
+from typing import Dict, Iterator, List, Union
 
 import tldextract
 
 from octopoes.models import OOI
+from octopoes.models.ooi.dns.records import NXDOMAIN
 from octopoes.models.ooi.dns.zone import Hostname
 from octopoes.models.ooi.email_security import DKIMExists
 from octopoes.models.ooi.findings import Finding, KATFindingType
 
 
-def run(input_ooi: Hostname, additional_oois: List[DKIMExists], config: Dict[str, str]) -> Iterator[OOI]:
+def run(
+    input_ooi: Hostname, additional_oois: List[Union[DKIMExists, NXDOMAIN]], config: Dict[str, str]
+) -> Iterator[OOI]:
+    dkim_exists = [ooi for ooi in additional_oois if isinstance(ooi, DKIMExists)]
+    nxdomains = [ooi for ooi in additional_oois if isinstance(ooi, NXDOMAIN)]
+
+    if nxdomains:
+        return
```

Comment:
Within this context is better to use a generator comprehension and use `any`, this way we don't have to go through the entire list to know if there is any `NXDOMAIN` instance within the list).
```suggestion
    nxdomains = (ooi for ooi in additional_oois if isinstance(ooi, NXDOMAIN))

    if any(nxdomains):
        return
```